### ISSUE TITLE
Polices

### ISSUE THREAD

**Issue by tvroylandt** [spyrales/gouvdown#5]
Récupérer les polices.

**Comment by tvroylandt** [comment]
J'ai vu que @RLesur avait ajouté la Spectrale en dépendance.
Est-ce que tu as moyen de mettre aussi la Marianne ?

**Comment by MaelTheuliere** [comment]
Je devrais récupérer Marianne bientôt via notre service informatique, reste la question de la diffusion de la police. J'espère que son statut le permettra.

**Comment by RLesur** [comment]
Salut, oui désolé, j'ai oublié de l'écrire ici. J'ai mis Spectral car sa licence (OSF) le permettait et qu'elle était disponible (logique vu qu'elle est open source). J'ai regardé le fichier de licence de Marianne : elle n'est pas compatible avec le logiciel libre car il y a une restriction d'usage (elle est réservée au seul usage du gouvernement).
Mettre Marianne dans le package poserait plein de difficultés de licence.
Je pense que les options sont : 1/  on suppose que l'utilisateur a installé Marianne et on utilise Arial en fallback font (mais ce sera galère pour mettre le résultat sur un serveur web) 2/ on se fait un package privé qu'on diffuse sous le manteau avec Marianne dedans et dans gouvdown, on checke discrètement s'il est installé (meilleure option selon moi).

**Comment by MaelTheuliere** [comment]
On commence par la solution 2 ?

**Comment by RLesur** [comment]
Je suis d'accord, je vous proposerais bien : 

- qu'on crée un package `gouvdown.fonts` (peut-être en mode privé) qui comprendrait les fontes et les déclarations CSS font face associées

- le package `gouvdown` repère discretos si `gouvdown.fonts` est installé : s'il n'est pas installé, on part du principe dans gouvdown qu'on s'appuie sur les fontes systèmes et s'il est installé, on utilise les fontes de `gouvdown.fonts`

Ca vous va ?

**Comment by RLesur** [comment]
J'ai créé : 

- le repository privé  `gouvdown.fonts` à l'adresse <https://github.com/spyrales/gouvdown.fonts/>

- une équipe @spyrales/gouvdown-fonts ayant accès en lecture à ce repository. De cette façon, pour accorder les droits en lecture, il suffira d'inclure de nouvelles personnes dans cette équipe.

**Comment by RLesur** [comment]
J'ai fait un premier jet qui devrait fonctionner et qu'on pourrait utiliser pour les autres polices, si cela vous convient: 

- les fichiers de la police Spectral vivent toutes dans [gouvdown.fonts](https://github.com/spyrale/gouvdown.fonts/)

- j'ai créé une fonction dans **gouvdown** qui va chercher les polices de **gouvdown.fonts** si le package est installé et sinon va taper Google Fonts, voir là https://github.com/spyrales/gouvdown/blob/23722a503ea1223f17976fe2615f377ff11cf7b4/R/fonts.R#L16-L26 et là https://github.com/spyrales/gouvdown/blob/8fb4c6e0111b02069b308c8b07917a11793affc0/inst/resources/fonts/spectral/links.html#L1-L2

Dites-moi si ça vous va.

**Comment by MaelTheuliere** [comment]
super @RLesur

**Comment by RLesur** [comment]
bon ben il n'y aura plus qu'à intégrer la police Marianne dans `gouvdown.fonts` quand vous parviendrez à mettre la main dessus (ça risque d'arriver à vous avant moi).

**Comment by RLesur** [comment]
J'ai intégré Marianne dans `gouvdown.fonts`. En revanche, je n'ai trouvé que quatre styles [ici](https://github.com/DISIC/design.numerique.gouv.fr/tree/master/src/assets/fonts/Marianne) : Light, Regular, Bold et Extra Bold.

Avez-vous accès aux autres styles (Thin et Medium) ?

**Comment by MaelTheuliere** [comment]
> 
> 
> J'ai intégré Marianne dans `gouvdown.fonts`. En revanche, je n'ai trouvé que quatre styles [ici](https://github.com/DISIC/design.numerique.gouv.fr/tree/master/src/assets/fonts/Marianne) : Light, Regular, Bold et Extra Bold.
> 
> Avez-vous accès aux autres styles (Thin et Medium) ?

J'ai récupéré les fonts manquantes en otf, y a un outil pour les convertir en ttf ?

**Comment by RLesur** [comment]
Je pense que oui mais je ne m'y connais pas assez.

Sinon, on peut tester avec que de l'OpenType. Le choix du format des fontes est toujours une grosse galère pour que ça fonctionne sur tous les systèmes/tous les navigateurs. De mon expérience, le format TrueType passe quasiment partout sauf que c'est un format non compressé (comme OpenType).

**Comment by tvroylandt** [comment]
Il y a un point que je ne comprends pas ici.

En gros, sur les polices, on construit des dépendances HTML et on dit d'installer les polices si elles ne le sont pas.
Dans le cas où elles ne sont pas installées, la dépendance HTML va les chercher (plus ou moins) si `goudown.fonts` est là.

Mais à quel moment on les importe pour que `ggplot2` puisse les utiliser puisqu'elles ne sont jamais installées automatiquement ?

**Comment by RLesur** [comment]
Effectivement, la logique est différente pour les dépendances HTML et la dépendance ggplot2 : 

- les htmldependencies ne vont être utiles que pour le CSS

- pour ggplot2, il faut en plus installer les fontes sur la machine pour qu'elles soient détectées par le système

Ce qui fait que par exemple, sur ma machine pro, la dépendance HTML/CSS fonctionne mais comme je n'ai pas le droit d'installer de nouvelles polices (et que les polices officielles n'ont pas été installées par l'administrateur de mon poste), je ne peux pas produire de graphique ggplot2 qui les utilise.

**Comment by tvroylandt** [comment]
Merci. C'est à peu près à quoi j'étais arrivé mais je voulais être sûr

**Comment by jengelaere** [comment]
Hello la team gouvdown, pour info les fonts Marianne ont été publiées sur Github par la DISIC : https://github.com/DISIC/design.numerique.gouv.fr/tree/master/src/assets/fonts/Marianne
La seule précaution prise est l'application d'une licence particulière pour ce dossier du projet

**Comment by RLesur** [comment]
Merci @jengelaere pour l'info ! Je suis quand même un peu surpris par leur formulation car ils n'ont pas le droit de redistribuer [ce répertoire](https://github.com/DISIC/design.numerique.gouv.fr/tree/master/src/assets/fonts/Inter) sous licence MIT : la fonte Inter est sous licence OFL (c'est du pinaillage).
Bon, s'ils le font, ça signifie qu'en mettant bien au carré **gouvdown.fonts** au niveau des licences, on peut se permettre de l'ouvrir également. Ce sera beaucoup plus simple, je vais m'en occuper.
